src/main/kotlin/com/gamingbacklog/api/gamingbacklogapi/clients/IGDBClient.kt

```diff
@@ -26,21 +26,20 @@ class IGDBClient {
   val authUri = "https://id.twitch.tv/oauth2/token"
   val baseUri = "https://api.igdb.com/v4/"
 
+
   /**
    * Generates a token
    */
   fun authenticate(): Credentials {
-    val request: HttpRequest = HttpRequest.newBuilder()
-      .uri(URI.create("$authUri?client_id=$clientId&client_secret=$clientSecret&grant_type=client_credentials"))
-      .POST(HttpRequest.BodyPublishers.noBody())
-      .build()
-    try {
-      val response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString())
-      return Gson().fromJson(response.body(), Credentials::class.java)
-    } catch (ex: Exception) {
-      ex.printStackTrace()
+    val response = externalAPICall.postExternalCall(
+      "$authUri?client_id=$clientId&client_secret=$clientSecret&grant_type=client_credentials"
+    )
+    return if (response.statusCode() == 200) {
+      Gson().fromJson(response.body(), Credentials::class.java)
+    } else {
+      println("Status Code: ${response.statusCode()}, Error: ${response.body()}")
```

Comment:
Definitely for a future milestone, but we should probably think about how we want to log errors at some point.

Reply:
yeah errors are kind of a mess in general. I'm fixing that up a little bit in the next story but we should have something down the line (milestone 3?) for that

Reply:
Sounds like a plan!